internal/formats/common/cyclonedxhelpers/format.go

```diff
@@ -57,18 +65,50 @@ func toComponent(p pkg.Package) cyclonedx.Component {
 	}
 }
 
+func lookupRelationship(ty artifact.RelationshipType) bool {
+	switch ty {
+	case artifact.OwnershipByFileOverlapRelationship:
```

Comment:
This PR is great! It naturally builds on the new relationship features we've been adding over the last few weeks. I think the only issue with this PR is the relationship type in this switch case. 

I'm still learning about CycloneDX spec details; does CycloneDX have any specific definitions on what a dependency is? That is, is a CycloneDX dependency a package build dependency? dev dependency? Must it be a package-to-package dependency? Or are package-to-file / file-to-package dependencies allowed?

Good news is that I think the right relationship types can start being added in https://github.com/anchore/syft/issues/572 . This issue is all about extending the package catalogers with the ability to raise up relationships that they find in their ecosystems. For example, the DPKG status file has fields for package dependencies (http://www.fifi.org/doc/libapt-pkg-doc/dpkg-tech.html/ch1.html) which means that the `pkg.DpkgMetadata` can now be extended to extract out this information (the `Depends` field at least) and have that cataloger raise up relationships between any two packages that were both found to be installed. (#572 is means to cover all ecosystems, not just DPKG, but figured I'd be more specific here)

The major addition in this issue is that _**one or more TBD relationship types will be added**_ to accommodate describing these dependencies here:

https://github.com/anchore/syft/blob/main/syft/artifact/relationship.go#L3-L12

The relationships that are there now don't seem to match with what a CycloneDX dependency is (I think), but the new relationship types being added will.

I think that means that the switch case here will be replaced with one or more new relationship types that more closely describe "package-to-package dependencies". I don't think it makes sense to merge this PR with the `OwnershipByFileOverlapRelationship` relationship described (unless I'm wrong about the intention of the CycloneDX dependency definition here) but with a different switch case in the near future it makes a ton of sense 👍 .

Reply:
@wagoodman CycloneDx dependencies describe how components depend on other components through a dependency graph that represents direct and transitive relationships. Generally I found syft code already does that for SPDX with the mentioned relationship type.

>  The relationships that are there now don't seem to match with what a CycloneDX dependency is (I think), but the new relationship types being added will.

Could you share which you think those new relationship types are ? I found that `OwnershipByFileOverlapRelationship` was more general-purpose type while `ContainsRelationship` is focused to SPDX files.

Reply:
Could you share some pointers where I could start adding support for these new types ? I am very concerned of using a CycloneDX file without having the minimum set of properties defined by the NTIA https://www.ntia.doc.gov/files/ntia/publications/sbom_minimum_elements_report.pdf.

Reply:
Hey, I've added some comments on https://github.com/anchore/syft/issues/572#issuecomment-1000412163 that could help here. 

TL;DR we'll need to add at least the following SPDX relationships (from https://spdx.github.io/spdx-spec/relationships-between-SPDX-elements/):
- `RUNTIME_DEPENDENCY_OF `
- `DEV_DEPENDENCY_OF `
- `BUILD_DEPENDENCY_OF `
- `DEPENDENCY_OF `

They would be added here: https://github.com/anchore/syft/blob/v0.34.0/syft/artifact/relationship.go#L3

```golang
RuntimeDependencyOfRelationship RelationshipType = "runtime-dependency-of"
DevDependencyOfRelationship RelationshipType = "dev-dependency-of"
BuildDependencyOfRelationship RelationshipType = "build-dependency-of"
DependencyOfRelationship RelationshipType = "dependency-of"
```

Once added, these should all map to the case statement in your PR that checks if the given relationship should translate to a CycloneDX dependency.

Issue https://github.com/anchore/syft/issues/572 is all about beginning to leverage these new relationships, as of now we don't have any catalogers that surface these dependencies as relationships.

Reply:
@wagoodman Okay, I'll update the PR with these changes. Thanks.